src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -1171,4 +1188,11 @@ protected boolean isPartitionBufferEmpty(final String pipeName) {
     }
     return false;
   }
+
+  private void handleCleanerThreadException() {
+    Exception potentialEx = failedCleanerException.get();
+    if (potentialEx != null) {
+      throw new ConnectException("Cleaner encountered an exception", potentialEx);
```

Comment:
is this the correct exception type we should use?

Reply:
Yes connectexception seems correct since that will kill the connector.